src/Persistence/Session.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace atk4\data\Persistence;
+
+/**
+ * Array persistence which will store model data in session.
+ */
+class Session extends Array_
+{
+    /**
+     * Session container key.
+     *
+     * @var string
+     */
+    protected $session_key = '__atk_session';
+
+    /**
+     * Constructor. Can pass array of data in parameters.
+     */
+    public function __construct(&$data = [], string $key = null)
+    {
+        $key = $key ?? $this->name ?? static::class;
+
+        parent::__construct($data);
+        $_SESSION[$this->session_key][$key] = &$this->data;
```

Comment:
you may need to do session_start at some point.

Reply:
Exactly, if not set to start by default... Shoul be simply never in data...

Reply:
At this point headers probably are already sent.

Reply:
It is very config dependent - autostart, session backend, buffering, ...